### ISSUE TITLE
Setting up TimedTL resets arrows and vice versa

### ISSUE THREAD

**Issue by kvakvs** [CitiesSkylinesMods/TMPE#961]
### Describe the problem

As reported by Biffa, setting up lane arrows and then traffic light, the arrows reset. Setting them again, leaves timed TL confused.

### Steps to reproduce

I want this junction to be set up like this (see the road coming from the bottom). I want no left hand turns as there's a roundabout off-screen so cars can turn and come back:
![bild](https://user-images.githubusercontent.com/288724/85869454-fbdc8a00-b7cb-11ea-95dc-2676172e5959.png)

But after setting this up, if I then use the quick setup of Time Lights it changes the lane to add back in the left turn:
![bild](https://user-images.githubusercontent.com/288724/85869478-04cd5b80-b7cc-11ea-9959-34d5f5a7ff1d.png)

then if I remove that left turn again, the traffic lights still think its there. I think it's a visual thing as the cars in that lane go straight on as required, even though the lights show it as a left hand turn
![bild](https://user-images.githubusercontent.com/288724/85869494-09920f80-b7cc-11ea-8758-e6638509e827.png)

**Comment by kianzarrin** [comment]
Where did Biffa report this? In his video? if so can you give me a link and time?

Setting up TImed traffic lights allocates dedicated turning lanes as it is required for the TTL to work. that's why the lanes change. We can add more clever logic to allocate dedicated turning lanes only toward already existing directions. Not sure if it is worth the effort.

> then if I remove that left turn again, the traffic lights still think its there. I think it's a visual thing as the cars in that lane go straight on as required, even though the lights show it as a left-hand turn

The important part is that it functionally works. but you can change the traffic light type as part of post modifications. Of course, it would be nice to automatically refresh the visual to when lane arrows change.

**Comment by kvakvs** [comment]
in #tmpe chat today at 13:14

**Comment by kianzarrin** [comment]
This is not a bug but a feature.

**Comment by kianzarrin** [comment]
https://discordapp.com/channels/545065285862948894/545066612202405889/726082285614661664
> mm, took a look and not sure what you mean by change mode option?
nvm found it!
yeah that makes no difference, still stays as a left turn on the lights even though it shouldnt :slight_smile:
although I can just turn it off with that button

if left traffic light shows even though there are no left lanes then this IS a bug! but the bug is with the GUI.

**Comment by kianzarrin** [comment]
Setting up traffic light changes lane arrows. this is a feature not a bug. Obiously there is a  need to allocate turning lanes based ont the Traffic Lights configuration.

@kvakvs Can you please modify the description. this is Traffic Lights GUI problem. I suggest that we close this and open a new one for the GUI to avoid confustion. Is that OK?

**Comment by kvakvs** [comment]
Sure, you're welcome to close this and reopen or edit.

**Comment by DaEgi01** [comment]
I'd rather see the TL adjust to the lane arrows .. and not the other way around. IMO its a bug :P

**Comment by aubergine10** [comment]
I agree with @DaEgi01 - a normally-created TTL should not, by default, be messing with my other road settings.

We could use the quick-setup TTL if we want dedicated turning lanes; see also: https://github.com/CitiesSkylinesMods/TMPE/issues/888#issuecomment-997512642

**Comment by kianzarrin** [comment]
> I'd rather see the TTL adjust to the lane arrows / lane connector .. and not the other way around. IMO its a bug :P

But if I do not adjust the lane arrows then TTL may not work all together because it requires dedicated turning lanes. How do you want to address that problem?

this does not happen when you setup TTL. only when you use auto-setup. and fixing lane arrows is a required part of that automatic process otherwise TTL will not work. 

when I setup high priority road I also bulk edit traffic rules from multiple different managers. why would TTL be any different?

**Comment by kianzarrin** [comment]
> I'd rather see the TTL adjust to the lane arrows / lane connector

how?

>  not the other way around

TTL quick setup does not touch lane connections. just lane arrows
